### PR TITLE
Adapt labels/annotations from pkg/common/operation to the new API group

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
Adapt labels/annotations from `pkg/common/operation` package to the new API group.

**Which issue(s) this PR fixes**:
Ref #1649

**Special notes for your reviewer**:

**Release note**:

```action use
Some labels and annotations from to the old API group have now equivalents in the new API group as follows:

`confirmation.garden.sapcloud.io/deletion` -> `confirmation.gardener.cloud/deletion`
`garden.sapcloud.io/createdBy` -> `gardener.cloud/created-by`
`project.garden.sapcloud.io/name` -> `project.gardener.cloud/name`
`namespace.garden.sapcloud.io/project` -> `namespace.gardener.cloud/project`
`shoot.garden.sapcloud.io/unhealthy` -> removed
`shoot.garden.sapcloud.io/expirationTimestamp` -> `shoot.gardener.cloud/expiration-timestamp`
`shoot.garden.sapcloud.io/status` -> `shoot.gardener.cloud/status`
`shoot.garden.sapcloud.io/operation` -> `gardener.cloud/operation`
`shoot.garden.sapcloud.io/sync-period` -> `shoot.gardener.cloud/sync-period`
`shoot.garden.sapcloud.io/ignore` -> `shoot.gardener.cloud/ignore`
`shoot.garden.sapcloud.io/tasks` -> `shoot.gardener.cloud/tasks`


The listed entries belonging to the old API group `garden.sapcloud.io` are now deprecated.
Make sure to adapt your codebase to the entries from the new API group `gardener.cloud`.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
I assume you will open another PR for the remaining labels?

File: plugin/pkg/shoot/quotavalidator/admission.go [hunk 1]
```diff
@@ -229,7 +229,7 @@ func (q *QuotaValidator) Validate(ctx context.Context, a admission.Attributes, o
 	}
 
 	// Admit Shoot lifetime changes
-	if lifetime, exists := shoot.Annotations[common.ShootExpirationTimestamp]; checkLifetime && exists && maxShootLifetime != nil {
+	if lifetime, exists := shoot.Annotations[common.ShootExpirationTimestampDeprecated]; checkLifetime && exists && maxShootLifetime != nil {
```
Comment (rfranzke) [inline plugin/pkg/shoot/quotavalidator/admission.go]: Why is here no handling for the new/non-deprecated expiration timestamp annotation?

**Review by rfranzke — COMMENTED** [review]
Overall, looks good!

File: plugin/pkg/shoot/quotavalidator/admission.go (hunk 1 above)
Comment (ialidzhikov) [reply plugin/pkg/shoot/quotavalidator/admission.go]: I was thinking about some corner cases about this annotation but I guess it is fine to have the handling for for the new/deprecated annotation. Thanks!

**Comment by ialidzhikov** [comment]
> I assume you will open another PR for the remaining labels?

Yes, my idea is to keep the PRs small to make them easy for review && validation.

File: docs/usage/shoot_operations.md [hunk 2]
```diff
@@ -6,35 +6,35 @@ Some of the operations can also not be caused by changing something in the shoot
 
 ## Immediate reconciliation
 
-Annotate the shoot with `shoot.garden.sapcloud.io/operation=reconcile` to make the `gardener-controller-manager` start a reconciliation operation without changing the shoot spec and possibly without being in its maintenance time window:
+Annotate the shoot with `gardener.cloud/operation=reconcile` to make the `gardener-controller-manager` start a reconciliation operation without changing the shoot spec and possibly without being in its maintenance time window:
```
Comment (timebertt) [inline docs/usage/shoot_operations.md]:
```suggestion
Annotate the shoot with `gardener.cloud/operation=reconcile` to make the `gardenlet` start a reconciliation operation without changing the shoot spec and possibly without being in its maintenance time window:
```

File: docs/usage/shoot_operations.md [hunk 3]
```diff
@@ -6,35 +6,35 @@ Some of the operations can also not be caused by changing something in the shoot
...
 
 ```bash
-kubectl -n garden-<project-name> annotate shoot <shoot-name> shoot.garden.sapcloud.io/operation=maintain
+kubectl -n garden-<project-name> annotate shoot <shoot-name> gardener.cloud/operation=maintain
 ```
 
 ## Retry failed operation
 
-Annotate the shoot with `shoot.garden.sapcloud.io/operation=retry` to make the `gardener-controller-manager` start a new reconciliation loop on a failed shoot.
+Annotate the shoot with `gardener.cloud/operation=retry` to make the `gardener-controller-manager` start a new reconciliation loop on a failed shoot.
```
Comment (timebertt) [inline docs/usage/shoot_operations.md]:
```suggestion
Annotate the shoot with `gardener.cloud/operation=retry` to make the `gardenlet` start a new reconciliation loop on a failed shoot.
```

File: docs/usage/shoot_operations.md [hunk 4]
```diff
@@ -6,35 +6,35 @@ Some of the operations can also not be caused by changing something in the shoot
...
 
 ```bash
-kubectl -n garden-<project-name> annotate shoot <shoot-name> shoot.garden.sapcloud.io/operation=retry
+kubectl -n garden-<project-name> annotate shoot <shoot-name> gardener.cloud/operation=retry
 ```
 
 ## Rotate kubeconfig credentials
 
-Annotate the shoot with `shoot.garden.sapcloud.io/operation=rotate-kubeconfig-credentials` to make the `gardener-controller-manager` exchange the credentials in your shoot cluster's kubeconfig.
+Annotate the shoot with `gardener.cloud/operation=rotate-kubeconfig-credentials` to make the `gardener-controller-manager` exchange the credentials in your shoot cluster's kubeconfig.
```
Comment (timebertt) [inline docs/usage/shoot_operations.md]:
```suggestion
Annotate the shoot with `gardener.cloud/operation=rotate-kubeconfig-credentials` to make the `gardenlet` exchange the credentials in your shoot cluster's kubeconfig.
```

File: pkg/controllerutils/miscelleneous.go [hunk 5]
```diff
@@ -26,8 +26,9 @@ const separator = ","
 // AddTasks adds a task to the ShootTasks annotation of the passed map.
 func AddTasks(existingAnnotations map[string]string, tasksToAdd ...string) {
 	var tasks []string
-	if len(existingAnnotations[common.ShootTasks]) > 0 {
-		tasks = strings.Split(existingAnnotations[common.ShootTasks], separator)
+	existingTasks, _ := common.GetTasksAnnotation(existingAnnotations)
```
Comment (timebertt) [inline pkg/controllerutils/miscelleneous.go:29]: BTW: typo in the filename.
Can you rename `pkg/controllerutils/miscelleneous.go` to `...miscellaneous.go`?

File: pkg/operation/common/types.go [hunk 6]
```diff
@@ -58,20 +58,28 @@ const (
 
 	// ConfirmationDeletion is an annotation on a Shoot resource whose value must be set to "true" in order to
 	// allow deleting the Shoot (if the annotation is not set any DELETE request will be denied).
-	ConfirmationDeletion = "confirmation.garden.sapcloud.io/deletion"
+	ConfirmationDeletion = "confirmation.gardener.cloud/deletion"
```
Comment (timebertt) [inline pkg/operation/common/types.go:61]: This is also used for confirming a `Project` deletion, right?
Can you adapt the doc string accordinly, please?

**Review by timebertt — CHANGES_REQUESTED** [review]
Nice PR, thanks for cleaning up :)
Just some small remarks, otherwise /lgtm

**Comment by ialidzhikov** [comment]
Very nice catches @tim-ebert ! Thank you!

File: pkg/operation/common/types.go [hunk 7]
```diff
@@ -271,23 +294,31 @@ const (
 	// ShootExpirationTimestamp is an annotation on a Shoot resource whose value represents the time when the Shoot lifetime
 	// is expired. The lifetime can be extended, but at most by the minimal value of the 'clusterLifetimeDays' property
 	// of referenced quotas.
-	ShootExpirationTimestamp = "shoot.garden.sapcloud.io/expirationTimestamp"
+	ShootExpirationTimestamp = "shoot.gardener.cloud/expiration-timestamp"
```
Comment (timebertt) [inline pkg/operation/common/types.go:297]: Can you please add a line to the release note for this change also?

File: pkg/operation/common/types.go [hunk 8]
```diff
@@ -298,7 +329,12 @@ const (
 	ShootOperationRotateKubeconfigCredentials = "rotate-kubeconfig-credentials"
 
 	// ShootTasks is a constant for an annotation on a Shoot which states that certain tasks should be done.
-	ShootTasks = "shoot.garden.sapcloud.io/tasks"
+	ShootTasks = "shoot.gardener.cloud/tasks"
```
Comment (timebertt) [inline pkg/operation/common/types.go:332]: Can you please add a line to the release note for this change also?

**Review by timebertt — APPROVED** [review]
Thanks for addressing my comments!
/approve

Please add the missing two label changes to the release note and then we are good to go ;)

**Comment by ialidzhikov** [comment]
> Please add the missing two label changes to the release note and then we are good to go ;)

Updated the release note.

**Pull request merged**